### DISCUSSION TITLE
3 post-install questions: collections, favicons, pagination

### DISCUSSION THREAD

**Discussion by sudorifix** [mmistakes/so-simple-theme#364] [Q&A] [answered]
Hi Michael & collaborators,
Firstly, *love* this theme, thank you for all the hard work & ongoing maintenance of it!
I've spun up a new site today using so simple as a remote theme: https://sudorifix.github.io/blog/ (repo: https://github.com/sudorifix/blog)
I've figured out & tinkered with a fair bit, despite being a n00b & feeling rather dim much of the time.
I have 3 remaining questions due to features that aren't quite resolved yet - apologies in advance if they're nuisance questions.

1. How can I get collection files (projects list) to display?
- I've adapted the recipes collection in your example site & popped a page file in it for testing. When it wouldn't work locally I dug through the layout files: I can see how the collection.html layout calls documents.collection.html & it's not tied to anything called recipes per se. I've not tinkered with the mechanism, but for some reason the file I've placed in the projects collection isn't appearing. Am I missing something?
2. How do I get the favicon to display?
- I've generated a set that display fine locally, but they're not present now the site is running from github; to get it working locally I've created a local /_includes/head-custom.html file with the favicon details as supplied with the package. It's not a browser (Firefox) caching issue, as I've checked alternative browsers (DuckDuckGo, Chrome), & the code itself seems fine according to other discussions I've seen on stackoverflow, so is the local head-custom file not overriding the remote_theme instructions?
3. How do I get pagination to display on the home page?
- I've turned it on in _config.yml & set paginate to true on the index.html file, plus played around with a lower limit on index.md (10 down to 5) when testing locally, however I've not succeeded in forcing pagination on the home layout like it appears on the posts layout. Do I need to add other front matter instructions to index.md?

Thanks in advance!
Jess

**Comment by mmistakes** [comment, marked as answer]
1. Adding collection .md files isn't enough. You need to configure Jekyll so it knows about it. Refer to [their documentation](https://jekyllrb.com/docs/collections/). If you look at the theme's test files you'll see these lines in [`_config.yml`](https://github.com/mmistakes/so-simple-theme/blob/master/example/_config.yml#L78-L82). This is how the Recipes collection is configured.

```yaml
collections:
  recipes:
    output: true
    permalink: /:collection/:path/
```

2. Your approach for adding favicons to a local `_includes/head-custom.html` file is correct. You'll want to examine your source, but on quick look the paths to the favicons are wrong and they are 404ing. That's why they don't show up for you. You need to add a `{{ site.baseurl }}` path before them or use Jekyll's `relative_url` filter on the paths so you get `/blog/apple-touch-icon.png` instead of the following:

```html
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#31ac94">
<meta name="msapplication-TileColor" content="#2b5797">
<meta name="theme-color" content="#31ac94">
```

3. You have two index (home page) files. `index.html` and `index.markdown`. Remove `index.markdown` and it should work. It's likely overriding the first one which is the version that enables pagination.

**Comment by sudorifix** [comment]
Thanks for the quick response, Michael! Much appreciated.
That all makes perfect sense, & has worked a treat so far... 
For the favicons, though, I've taken the logic one step further & copied the icon files to a subdirectory of /images. With relative urls specified in the head (pointing to the subdirectory) it's working fine at my end, but can you foresee any problems with collecting icons from somewhere other than the root?
Also, my sample collection/project page is displaying raw text only. Should it not pick up the collection.html layout instructions sourced from the remote theme?

**Comment by mmistakes** [comment]
Paths shouldn't matter, as long as they resolve you'll be fine.

For your collection pages, you sure you've set the `layout` in the YAML front matter?

**Comment by sudorifix** [comment]
Sorry Michael, work’s been crazy busy. No page layout spec in the yaml; I copied the file from the theme’s example (https://github.com/mmistakes/so-simple-theme/blob/master/example/_recipes/peanut-butter-cookies.md) without making a change, to see what it’d do. No problems adding it, makes sense, but still curious how the original displayed properly but once copied to my repo it didn’t?

**Comment by mmistakes** [comment]
Hard to tell what went wrong for you without seeing your actual site files.
My guess, you missed some configs. Collection documents need to be configured, they don't "just work" by dropping a file into a `_underscore` folder. Only `_posts` work that way.
